### ISSUE TITLE
negative_edge_cycle calling bellman_ford with mismatched argument

### ISSUE THREAD

**Issue by shichuzhu** [networkx/networkx#4657]
### Current Behavior
[`negative_edge_cycle`](https://github.com/networkx/networkx/blob/7a1b50e5b4c348aa8b83b0a852aa25439c7dc83a/networkx/algorithms/shortest_paths/weighted.py#L1911) function cannot handle custom input argument `weight` correctly, because the [`weight`](https://github.com/networkx/networkx/blob/7a1b50e5b4c348aa8b83b0a852aa25439c7dc83a/networkx/algorithms/shortest_paths/weighted.py#L1968) argument is propagated to [`bellman_ford_predecessor_and_distance`](https://github.com/networkx/networkx/blob/7a1b50e5b4c348aa8b83b0a852aa25439c7dc83a/networkx/algorithms/shortest_paths/weighted.py#L1111) accidentally as `target` instead of `weight`.

```python
from networkx import negative_edge_cycle, DiGraph

d = DiGraph()
d.add_node('a')
d.add_node('b')
d.add_edge('a', 'b', w=-2)
d.add_edge('b', 'a', w=-1)
print(negative_edge_cycle(d, weight='w')) # outputs False
```

### Expected Behavior
Expect outputs True

### Steps to Reproduce
see above python script

### Environment
Python version: 3.9.2
NetworkX version: 2.6rc1.dev0

**Comment by shichuzhu** [comment]
Note this seems an obvious thing to fix and I'm working on a PR right now.
I'm just going to use keyword argument instead of positional argument.

**Comment by dschult** [comment]
Thanks!

**Comment by rossbar** [comment]
Closed by #4658
